### PR TITLE
DHFPROD-5950: Exploring curated entities from staging and final databases

### PR DESCRIPTION
**Pull request by rahulvudutala**
# Description

#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

**Review by ryanjdew — DISMISSED** [review]

**Review by timur-isangulov — APPROVED** [review]
Tested it while working on UI part, looks good.

File: marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/controllers/EntitySearchController.java [hunk 1]
```diff
@@ -138,31 +138,40 @@ public JsonNode getFacetValuesRange(@RequestBody @ApiParam(hidden = true) JsonNo
...
+    private EntitySearchManager newEntitySearchManager(String database) {
+        return new EntitySearchManager(getHubClient(), database);
     }
 
     private EntitySearchService getEntitySearchService() {
+        return getEntitySearchService("final");
+    }
+
+    private EntitySearchService getEntitySearchService(String database) {
+        if(database.equalsIgnoreCase("staging")) {
```
Comment (rjrudin) [inline marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/controllers/EntitySearchController.java:172]: To ensure an NPE is never possible here, do "staging".equalsIgnoreCase(database)

File: marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/entities/search/EntitySearchManager.java [hunk 2]
```diff
@@ -79,17 +76,31 @@
...
 
     public EntitySearchManager(HubClient hubClient) {
-        this.finalDatabaseClient = hubClient.getFinalClient();
+        this.databaseClient = hubClient.getFinalClient();
+        this.modelManager = new ModelManager(hubClient);
+        initializeFacetHandlerMap();
+    }
+
+    public EntitySearchManager(HubClient hubClient, String database) {
+        if(database.equalsIgnoreCase("staging")) {
```
Comment (rjrudin) [inline marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/entities/search/EntitySearchManager.java:90]: Same thing, "staging".equals

File: marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/entities/search/EntitySearchManager.java [hunk 3]
```diff
@@ -79,17 +76,31 @@
 
     public static String QUERY_OPTIONS = "exp-final-entity-options";
     private static Map<String, FacetHandler> facetHandlerMap;
-    private DatabaseClient finalDatabaseClient;
+    private DatabaseClient databaseClient;
```
Comment (rjrudin) [inline marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/entities/search/EntitySearchManager.java:79]: The below variable has a really nice name, which raises the question of - what's this client used for? I think it'd be good to name this "searchDatabaseClient" or something similar to convey what this one is used for.

**Review by rjrudin — CHANGES_REQUESTED** [review]
Looks good, just a few small tweaks

**Review by rjrudin — APPROVED** [review]

**Review by akshaysonvane — APPROVED** [review]

**Review by timur-isangulov — APPROVED** [review]

**Review by xnikhil08 — APPROVED** [review]

**Pull request merged**